api/access_key.go

```diff
@@ -39,3 +49,20 @@ type CreateAccessKeyResponse struct {
 	ExtensionDeadline Time   `json:"extensionDeadline" note:"the key must be used by this time to remain valid"`
 	AccessKey         string `json:"accessKey"`
 }
+
+// ValidateName returns a standard validation rule for all name fields. The
+// field name must always be "name".
+func ValidateName(value string) validate.ValidationRule {
+	return validate.StringRule{
+		Value:     value,
+		Name:      "name",
+		MinLength: 3,
+		MaxLength: 256,
+		CharacterRanges: []validate.CharRange{
+			validate.AlphabetLower,
+			validate.AlphabetUpper,
+			validate.Numbers,
+			validate.Dash, validate.Underscore, validate.Dot,
+		},
```

Comment:
This is an important piece to review. I'd like to apply this same validation to all name fields.

Does a length of 3-256 characters seem reasonable?
This will restrict the characters to `a-z`, `A-Z`, `0-9`, and `-`, `_`, `.`.

Reply:
We might need to allow the `@` character as well, for users, because I think we use the name field for email addresses.